examples/traceresponse/src/server.rs

```diff
@@ -0,0 +1,58 @@
+use hyper::service::{make_service_fn, service_fn};
+use hyper::{Body, Request, Response, Server};
+use opentelemetry::propagation::TextMapPropagator;
+use opentelemetry::trace::FutureExt;
+use opentelemetry::{
+    global,
+    sdk::export::trace::stdout,
+    sdk::{
+        propagation::TraceContextPropagator,
+        trace::{self, Sampler},
+    },
+    trace::{Span, Tracer},
+};
+use opentelemetry_contrib::trace::propagator::trace_context_response::TraceContextResponsePropagator;
+use opentelemetry_http::{HeaderExtractor, HeaderInjector};
+use std::{convert::Infallible, net::SocketAddr};
+
+async fn handle(req: Request<Body>) -> Result<Response<Body>, Infallible> {
+    let parent_cx = global::get_text_map_propagator(|propagator| {
+        propagator.extract(&HeaderExtractor(req.headers()))
+    });
+    let _cx_guard = parent_cx.attach();
+    let mut span = global::tracer("example/server").start("hello");
+    span.add_event("handling this...", Vec::new());
+
+    let mut res = Response::new("Hello, World!".into());
+
+    let response_propagator: &dyn TextMapPropagator = &TraceContextResponsePropagator::new();
+    response_propagator.inject(&mut HeaderInjector(res.headers_mut()));
```

Comment:
(Rust newbie comment) - this is supposed to return the traceid-spanid-traceflags, where spanid will be the spanid of the server span created here, right? I do not see this setting the right spanid here...

Reply:
Doh. You're right 🤦 that's on me for going too fast, will fix 👍